File: presentation/src/main/java/com/github/anrimian/musicplayer/ui/settings/player/PlayerSettingsView.java
package com.github.anrimian.musicplayer.ui.settings.player;

import moxy.MvpView;
import moxy.viewstate.strategy.AddToEndSingleStrategy;
import moxy.viewstate.strategy.StateStrategyType;

public interface PlayerSettingsView extends MvpView {

    @StateStrategyType(AddToEndSingleStrategy.class)
    void showDecreaseVolumeOnAudioFocusLossEnabled(boolean checked);

    @StateStrategyType(AddToEndSingleStrategy.class)
    void showSelectedEqualizerType(int type);
}

File: presentation/src/main/java/com/github/anrimian/musicplayer/ui/settings/player/PlayerSettingsView.kt
package com.github.anrimian.musicplayer.ui.settings.player

import moxy.MvpView
import moxy.viewstate.strategy.alias.AddToEndSingle

interface PlayerSettingsView : MvpView {

    @AddToEndSingle
    fun showDecreaseVolumeOnAudioFocusLossEnabled(checked: Boolean)

    @AddToEndSingle
    fun showPauseOnAudioFocusLossEnabled(checked: Boolean)

    @AddToEndSingle
    fun showPauseOnZeroVolumeLevelEnabled(enabled: Boolean)

    @AddToEndSingle
    fun showSelectedEqualizerType(type: Int)

    @AddToEndSingle
    fun showEnabledMediaPlayers(players: IntArray)

}